File: server/schema.go
package main

type ExecuteCodeRequest struct {
	Code     string            `json:"code" validate:"required"`
	Language SupportedLanguage `json:"language" validate:"required"`
}

type ExecuteCodeResponse struct {
	Output      string `json:"output"`
	TimeElapsed int    `json:"time_elapsed"`
}

type SupportedLanguage string

const (
	Python SupportedLanguage = "python"
	Go     SupportedLanguage = "go"
)

File: client/src/lib/schema.ts
import { Problem } from "@/mongo/schema"

export interface ExecuteCodeResponse {
  output: string
  time_elapsed: number
}

export interface ExecuteCodeRequest {
  code: string
  language: string
}

export interface GetProblemRequest {
  problemId: string
}

export interface GetProblemResponse {
  markdownHtml: string
}

export interface InsertProblemRequest {
  problem: Problem
}
export interface InsertProblemResponse {
  problemId: string
}
